src/git-prune-branches

```diff
@@ -22,8 +22,23 @@ function should_delete_branch {
 }
 
 
+function set_previous_branch {
+  # previous branch was deleted
+  if [ "$prune_branches_initial_previous_branch_deleted" = true ]; then
+    set_branch_as_previous_branch "$MAIN_BRANCH_NAME"
+  # current branch was deleted
+  elif [ "$prune_branches_initial_branch_deleted" = true ]; then
+    set_branch_as_previous_branch "$INITIAL_PREVIOUS_BRANCH_NAME"
+  # some other branch was deleted
+  else
+    set_branch_as_previous_branch "$INITIAL_PREVIOUS_BRANCH_NAME"
+  fi
+}
+
+
 function steps {
-  local initial_branch_deleted=false
+  prune_branches_initial_branch_deleted=false
+  prune_branches_initial_previous_branch_deleted=false
```

Comment:
why are these namespace? initial_branch_deleted and initial_previous_branch_deleted (i think it should just be previous_branch_deleted) is sufficient.